detectors/aws/eks/detector.go

```diff
@@ -139,50 +139,34 @@ func (eksUtils eksDetectorUtils) fileExists(filename string) bool {
 	return err == nil && !info.IsDir()
 }
 
-// fetchString executes an HTTP request with a given HTTP Method and URL string.
-func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string) (string, error) {
-	request, err := http.NewRequest(httpMethod, URL, nil)
-	if err != nil {
-		return "", fmt.Errorf("failed to create new HTTP request with method=%s, URL=%s: %w", httpMethod, URL, err)
-	}
-
-	// Set HTTP request header with authentication credentials
-	authHeader, err := getK8sCredHeader()
+// fetchString executes an HTTP Kubernetes client request with a given HTTP Method and URL string.
+func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string, API string) (string, error) {
+	// Get cluster configuration
+	confs, err := getClusterConfig(URL)
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to create config with method=%s, URL=%s: %w", httpMethod, URL, err)
 	}
-	request.Header.Set("Authorization", authHeader)
 
-	// Get certificate
-	caCert, err := ioutil.ReadFile(k8sCertPath)
+	// Create clientset using generated configuration
+	clientset, err := kubernetes.NewForConfig(confs)
 	if err != nil {
-		return "", fmt.Errorf("failed to read file with path %s", k8sCertPath)
+		return "", fmt.Errorf("failed to create clientset for Kubernetes client")
```

Comment:
It seems like you're overwriting the error here?
```suggestion
		return "", fmt.Errorf("failed to create clientset for Kubernetes client: %w", err)
```
You could also add in the `confs` passed to `newForConfig` into the error message?

Reply:
Regarding the overwriting part, the functionality is a little different from earlier that is why I adjusted the error message accordingly. 

I can add the `err` into the message. But adding `confs` may make it reduce the readability so I'm not sure if I add that.